hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/ozoneimpl/TestOzoneContainer.java

```diff
@@ -158,6 +161,134 @@ public void testOzoneContainerViaDataNode() throws Exception {
     }
   }
 
+  @Test
+  public void testOzoneContainerWithMissingContainer() throws Exception {
+    MiniOzoneCluster cluster = null;
+    try {
+      long containerID =
+          ContainerTestHelper.getTestContainerID();
+      OzoneConfiguration conf = newOzoneConfiguration();
+
+      // Start ozone container Via Datanode create.
+      cluster = MiniOzoneCluster.newBuilder(conf)
+          .setNumDatanodes(1)
+          .build();
+      cluster.waitForClusterToBeReady();
+
+      runTestOzoneContainerWithMissingContainer(cluster, containerID);
+    } finally {
+      if (cluster != null) {
+        cluster.shutdown();
+      }
+    }
+  }
+
+  private void runTestOzoneContainerWithMissingContainer(
+      MiniOzoneCluster cluster, long testContainerID) throws Exception {
+    ContainerProtos.ContainerCommandRequestProto
+        request, writeChunkRequest, putBlockRequest,
+        updateRequest1, updateRequest2;
+    ContainerProtos.ContainerCommandResponseProto response,
+        updateResponse1, updateResponse2;
+    XceiverClientGrpc client = null;
+    try {
+      // This client talks to ozone container via datanode.
+      client = createClientForTesting(cluster);
+      client.connect();
+      Pipeline pipeline = client.getPipeline();
+      createContainerForTesting(client, testContainerID);
+      writeChunkRequest = writeChunkForContainer(client, testContainerID,
+          1024);
+
+      DatanodeDetails datanodeDetails = cluster.getHddsDatanodes().get(0).getDatanodeDetails();
+      File containerPath =
+          new File(cluster.getHddsDatanode(datanodeDetails).getDatanodeStateMachine()
+              .getContainer().getContainerSet().getContainer(testContainerID)
+              .getContainerData().getContainerPath());
+      cluster.getHddsDatanode(datanodeDetails).stop();
+      FileUtils.deleteDirectory(containerPath);
+
+      // Restart & Check if the container has been marked as missing, since the container directory has been deleted.
+      cluster.restartHddsDatanode(datanodeDetails, false);
+      GenericTestUtils.waitFor(() -> {
+        try {
+          return cluster.getHddsDatanode(datanodeDetails).getDatanodeStateMachine()
+              .getContainer().getContainerSet()
+              .getMissingContainerSet().contains(testContainerID);
+        } catch (IOException e) {
+          return false;
+        }
+      }, 1000, 30000);
+
+      // Read Chunk
```

Comment:
Should we add a second write chunk as well?

Reply:
done

Reply:
I have added a loop for testing creating open container. Then post restarting I am creating a recovering container and performing the same set operations expecting it to succeed.

Reply:
I am testing both creating an open container and recovering container. Agnostic of whether it is an EC key or a ratis key we land in the same dispatcher flow. So we don't have to test this separately https://github.com/apache/ozone/blob/827bc86915efbb398f024ccc467de5db7ecd347d/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/ozoneimpl/TestOzoneContainer.java#L253-L256

Reply:
in a follow up jira please accept write chunks only for EC pipelines if container is recovering.

Reply:
I have created a follow up jira for performing this validation:
https://issues.apache.org/jira/browse/HDDS-11761